randovania/game_connection/connector/prime_remote_connector.py

```diff
@@ -270,6 +270,16 @@ async def receive_remote_pickups(
         patches.append(self._dol_patch_for_hud_message(message))
 
         await self.execute_remote_patches(patches)
+
+        # Safety/debugging check to see if this is the spot where the magic item randomly increases by 2 instead of 1.
+        expected_capacity = magic_inv.capacity + 1
+        memory_ops = await self._memory_op_for_items([multiworld_magic_item])
+        op_result = await self.executor.perform_single_memory_operation(*memory_ops)
+        assert op_result is not None
+        magic_inv = InventoryItem(*struct.unpack(">II", op_result))
+        if magic_inv.capacity != expected_capacity:
+            self.logger.warning(f"Magic capacity should be '{expected_capacity}' but it was '{magic_inv.capacity}'")
```

Comment:
Good chance this will always trigger on Dolphin. Probably not on Nintendont because network slow enough for a new frame to happen.

When `execute_remote_patches` ends, all it means we've finished writing to game memory the code to run. At the earliest it'll run next frame for the game.

Reply:
ah. meh. reworked to check after the game has ran the code